### DISCUSSION TITLE
Scale value limitation

### DISCUSSION THREAD

**Discussion by DarthSim** [libvips/libvips#2115] [Q&A]
Hey @jcupitt!

I'm just wondering if there's any reason to limit the min value of [scale](https://github.com/libvips/libvips/blob/master/libvips/foreign/svgload.c#L436-L441) from `vips_svgload` and `vips_pdfload`. We bumped into the issue because of this limitation - https://github.com/imgproxy/imgproxy/issues/550. I managed to bypass this by reducing `dpi`, but still. Actually, the fact that I managed to bypass the limitation and everything still works shows the fact that the limitation is kinda redundant.

**Comment by jcupitt** [comment]
Hi @DarthSim, I think the range limits are just a sanity check -- it's good to detect crazy values early.

What range do you think would be better?

**Reply by DarthSim** [reply]
Sanity checks are a good thing but in this case, IMHO they work wrong as they don't prevent the operation but the operation is executed in an unexpected way. In our case, this leads to OOM.

I'm not sure which values can be called "crazy" here, but since the max value is `100000.0` I think the min value of `0.000001` should work too.

Also, it'd be good to add a check to prevent zero width/height. If you're ok with changing the min value, I could prepare a PR.

**Reply by jcupitt** [reply]
That would be great! Thanks very much.

Yes, I hadn't anticipated strange SVGs like the one in that linked issue.
